sdk/keyvault/keyvault-admin/src/lro/backup/operation.ts

```diff
@@ -129,12 +129,10 @@ export class BackupPollOperation extends KeyVaultAdminPollOperation<
       state.statusDetails = statusDetails;
       state.result = azureStorageBlobContainerUri;
 
-      if (endTime) {
-        state.isCompleted = true;
-      }
-      if (error && error.message) {
-        state.isCompleted = true;
-        state.error = new Error(error.message);
+      state.isCompleted = !!(endTime || error?.message);
+
+      if (error?.message || statusDetails) {
+        state.error = new Error(error?.message || statusDetails);
```

Comment:
Thank you!

Reply:
This error detection seems to be the same across LROs for this client.

It seems brittle to me that we have these same conditionals in different places. If it seems brittle to you to, I wonder if we can we move this to be at the root of the lro/ folder? It could be either in the existing file there, or in a new file. I'm thinking something we could use in the shape of:

```ts
state.isCompleted = checkCompelted(serviceOperation);
state.error = getError(serviceOperation);
```

I'm not sure about the names. It could also be some method of the parent class `KeyVaultAdminPollOperation` to update the state based on common logic, so in that case it could end up being used like:

```ts
state = this.handleAdminResponse(serviceOperation, state);
```

I can think of many different approaches that would be equally valid for me. I'm hoping the idea makes sense! If so, please do as you may. I trust your instinct.

Reply:
Thanks for the feedback! This is good advice, and one that I considered as well. I want to add a `mapStatus` method to the base poller but I need to get a better handle on the usages and think about what makes sense. So what I'll do is create an issue and assign it to myself to refactor this logic to the `keyVaultAdminPoller`.

Reply:
Please refrence the mapStatus here when you have the time to make the issue 🌞

Reply:
Yup, created #13321